tests/lib/gladecheck.py

```diff
@@ -78,107 +56,16 @@ def checkGlade(self, glade_tree):
-        super().configure(options, conf)
-
-        # If no glade files were specified, find all of them
-        if options.glade_file:
-            glade_files = options.glade_file
-        else:
-            glade_files = testfilelist(lambda x: x.endswith('.glade'))
-
-        # Parse all of the glade files
-        log.info("Parsing glade files...")
-        for glade_file in glade_files:
-            self.glade_trees.append(etree.parse(glade_file))
-
-        if options.translate:
-            log.info("Loading translations...")
-            podicts = translate_all(options.podir)
-
-            # Loop over each available language
-            for lang, langmap in podicts.items():
-                self.translated_trees[lang] = []
-
-                # For each language, loop over the parsed glade files
-                for tree in self.glade_trees:
-                    # Make a copy of the tree to translate and save it to
-                    # the list for this language
-                    tree = copy.deepcopy(tree)
-                    self.translated_trees[lang].append(tree)
-
-                    # Save the language as an attribute of the root of the tree
-                    tree.getroot().set("lang", lang)
-
-                    # Look for all properties with translatable=yes and translate them
-                    for translatable in tree.xpath('//property[@translatable="yes"]'):
-                        try:
-                            xlated_text = langmap.get(translatable.text, context=translatable.get('context'))[0]
-
-                            # Add the untranslated text as an attribute to this node
-                            translatable.set("original_text", translatable.text)
-
-                            # Replace the actual text
-                            translatable.text = xlated_text
-                        except KeyError:
-                            # No translation available for this string in this language
-                            pass
-
-    def prepareTestCase(self, testcase):
-        # Add the glade files to the GladeTest object
-        testcase.test.glade_trees = self.glade_trees
-        testcase.test.translated_trees = self.translated_trees
-
-    def describeTest(self, testcase):
-        # Return the first line of the doc string on checkGlade instead
-        # of the string for test_glade_file. If there is no doc string,
-        # return the name of the class.
-        doc = testcase.test.checkGlade.__doc__
-        if doc:
-            return doc.strip().split("\n")[0].strip()
-        else:
-            return testcase.test.__class__.__name__
 
-    def wantClass(self, cls):
-        # Make sure we grab all the GladeTest subclasses, and only GladeTest
-        # subclasses, regardless of name.
-        return issubclass(cls, GladeTest) and cls != GladeTest
+def load_glade_trees_from_files():
+    glade_files = testfilelist(lambda x: x.endswith('.glade'))
```

Comment:
Are you sure this is working? Usually, you need to use `global` to declare that this is a global variable.

Reply:
The tests actually take noticeable time (hundreds on ms) when run... But your idea below is better.